datadog/resource_datadog_dashboard.go

```diff
@@ -6463,3 +6498,15 @@ func validateTimeseriesWidgetLegendSize(val interface{}, key string) (warns []st
 	}
 	return
 }
+
+func validateTimeseriesWidgetLegendColumn(val interface{}, key string) (warns []string, errs []error) {
```

Comment:
We can move this function into the validators package: https://github.com/DataDog/terraform-provider-datadog/tree/master/datadog/internal/validators

Reply:
Hmm looks like this `LegendColumns` should have been an enum (see https://github.com/DataDog/datadog-api-spec/pull/870/files#diff-dd474d3ab204ed0a3b54a87c593cea4ea8cb4de62c65eaab653889e6d3cfda7eR2685-R2696), so I say, we should fix the spec to make it so.
And then use the default enum validate func

Reply:
I merged the fix in the client, can you update your version and fix here to use the enum ?

Reply:
@sighrobot I just committed the change, hope you don't mind.